app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/LoadInStorePickupOnGetPlugin.php

```diff
@@ -20,17 +20,16 @@ class LoadInStorePickupOnGetPlugin
      * @param SourceInterface $source
      *
      * @return SourceInterface
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function afterGet(
         SourceRepositoryInterface $subject,
         SourceInterface $source
     ):SourceInterface {
-        $pickupAvailable = $source->getData(InStorePickupInterface::IN_STORE_PICKUP_CODE);
+        $pickupAvailable = $source->getData(PickupLocationInterface::IS_PICKUP_LOCATION_ACTIVE);
 
         $extensionAttributes = $source->getExtensionAttributes();
-        $extensionAttributes->setInStorePickup($pickupAvailable);
-
-        $source->setExtensionAttributes($extensionAttributes);
+        $extensionAttributes->setIsPickupLocationActive((bool)$pickupAvailable);
```

Comment:
According to the `SourceInterface` contract `getExtensionAttributes` may also return null.

Reply:
Added check for `null`.